gobblin-core/src/main/java/org/apache/gobblin/publisher/HiveRegistrationPublisher.java

```diff
@@ -157,7 +146,9 @@ public void publishData(Collection<? extends WorkUnitState> states) throws IOExc
           if (isPathDedupeEnabled && pathsToRegisterFromSingleState.contains(path)){
             continue;
           }
-          pathsToRegisterFromSingleState.add(path);
+          if(isPathDedupeEnabled) {
```

Comment:
can this if () {} block be merged inside the above if() {} block? e.g. if (isPathDedupeEnabled) { pathsToRegisterFromSingleState.contains(path) ? continue: pathsToRegisterFromSingleState.add(path);}